src/scripts/main.js

```diff
@@ -1,3 +1,248 @@
 'use strict';
 
-// write code here
+// start everything
+initSorting();
+initSelection();
+initEditing();
+initFormAddingAndWorking();
+
+// init sorting functionality
+function initSorting() {
+  // init necessary helpfull data
+  const tbody = document.querySelector('tbody');
+  const ascOrders = {
+    name: true,
+    position: true,
+    office: true,
+    age: true,
+    salary: true,
+  };
+
+  // init heplhull functions
+  function appendListToTbody(list) {
+    list.forEach(tr => {
+      tbody.append(tr);
+    });
+  }
+
+  function setOtherToDefault(exception) {
+    for (const key in ascOrders) {
+      if (key !== exception) {
+        ascOrders[key] = true;
+      }
+    }
+  }
+
+  // init sorting sunctions
+  function sortNumbers(order, columnName) {
+    const list = [...tbody.querySelectorAll('tr')];
+
+    if (ascOrders[columnName]) {
+      list.sort((a, b) => a.children[order].innerText
+        - b.children[order].innerText);
+    } else {
+      list.sort((a, b) => b.children[order].innerText
+        - a.children[order].innerText);
+    }
+    ascOrders[columnName] = !ascOrders[columnName];
+    setOtherToDefault(columnName);
+
+    appendListToTbody(list);
+  }
+
+  function sortStrings(order, columnName) {
+    const list = [...tbody.querySelectorAll('tr')];
+
+    if (ascOrders[columnName]) {
+      list.sort((a, b) =>
+        a.children[order].innerText.localeCompare(b.children[order].innerText));
+    } else {
+      list.sort((a, b) =>
+        b.children[order].innerText.localeCompare(a.children[order].innerText));
+    }
+    ascOrders[columnName] = !ascOrders[columnName];
+    setOtherToDefault(columnName);
+
+    appendListToTbody(list);
+  }
+
+  function sortSalaries(order, columnName) {
+    function getNumberFromSalaryString(str) {
+      return +str.toString().slice(1).split(',').join('');
+    }
+
+    const list = [...tbody.querySelectorAll('tr')];
+
+    if (ascOrders[columnName]) {
+      list.sort((a, b) => getNumberFromSalaryString(a.children[order].innerText)
+      - getNumberFromSalaryString(b.children[order].innerText));
+    } else {
+      list.sort((a, b) => getNumberFromSalaryString(b.children[order].innerText)
+      - getNumberFromSalaryString(a.children[order].innerText));
+    }
+    ascOrders[columnName] = !ascOrders[columnName];
+
+    setOtherToDefault(columnName);
+    appendListToTbody(list);
+  }
+
+  // init events
+  document.querySelector('tr').addEventListener('click', (event) => {
+    switch (event.target.innerText) {
+      case 'Name': sortStrings(0, 'name'); break;
+      case 'Position': sortStrings(1, 'position'); break;
+      case 'Office': sortStrings(2, 'office'); break;
+      case 'Age': sortNumbers(3, 'age'); break;
+      case 'Salary': sortSalaries(4, 'salary'); break;
```

Comment:
usually cases are written on the separate lines

Reply:
here are more readable